### ISSUE TITLE
Unexpected shutdowns causing loss of handle!

### ISSUE THREAD

**Issue by FoxxZenovka** [illuminatedwax/pesterchum#12]
My computer unexpectedly shut off all of a sudden, and when I tried to restart it with my current handle, it told me that it was suddenly unavailable..! I'm not sure if this is just user stupidity on my end, but I thought I might as well report this. The handle was binaryTheif.

**Comment by chronicRapture** [comment]
i have this same problem! no one will answer me! and calsprite is useless.

**Comment by chronicRapture** [comment]
so this guy called http://www.mspaforums.com/member.php?1011876-ashdenej tells me what i did wrong. but he didnt tell me if my account was a lost cause or not. idk if this will help with your computer shutting down all of a sudden but.....oh well. maybe you'll have to make a new chumhandle...
http://www.mspaforums.com/showthread.php?37333-PESTERCHUM-3.14.1-New-Pesterchum-(6-26-3.41-8eta-pg54)/page68

**Comment by chronicRapture** [comment]
ok, so i think i know what to do. i think this is only if you have windows 7:
http://s1221.photobucket.com/albums/dd470/oceanpanda6/?action=view&current=Capture3.png
if you dont have the pesterchum icon there, go to the small ^ arrow thats pointing up and see if the icon is there. i just moved the icon from there to where it is in the picture. if its not there at all, then idk what to tell you. if it is, click it and your handle should come up. if there are multiple icons, click them and check them, then for the ones you dont want, right click the icons and press exit. they should go away. i hope this helps.

**Comment by ashen-dawn** [comment]
I had this issue too, but I found out how to fix it (on Ubuntu at least, but I've made up as good of instructions as I can from intuition and inference).

As far as I can tell the problem is caused when PesterChum exits unexpectedly and it fails to shut down the python scripts that are currently running through it.  That means the script that is waiting for responses from the IRC server is still up, so the IRC server thinks that your client is still connected (because technically a part of it still is).  That means that nobody can log on with that "nick" ("handle" to those who don't know pesterchum runs on IRC).  To fix this you need to forcefully end that python script.

In Windows: Press [Ctrl]+[Alt]+[Del].  If a menu comes up, then press the button that will open task manager.  In task manager you go to the processes tab (or something similar, I'm not on my windows computer at the moment) and find the process called Python <Insert version number here>.  Close that process BUT PESTERCHUM CANNOT BE OPEN WHILE YOU CLOSE IT OR YOU MIGHT BREAK SOMETHING.  Start up pesterchum again and it should be fixed.  Don't hold me to that, I'm not a developer here, just somebody who might possibly know what he's talking about.  If that doesn't work, maybe shutting down and booting-up your computer will fix it because it will close any programs that might still be activating that IRC connection.

In Linux: If you're using linux, then you should be competent enough to close python using the System Monitor.  If not...I'm using Ubuntu 11.04 with a Gnome 2 desktop system, and you're welcome to try what I tried.  Go to the top toolbar. System>Administration>System Monitor: Processes tab: Find Python.  Close it while pesterchum isn't open.  Similar to Windows...shutting down and a boot-up might also fix the problem.

On a Mac: I dunno.  They're pretty machines but I'm clueless on those.  It should be similar once you open your system monitor, and as always a total shut-down followed by a boot-up should also fix the problem.

EDIT: Please let me know if this works.  I'm anxious to know if my solution helped at all.

**Comment by kiooeht** [comment]
Have no fear, your chumhandle is not lost, no need to create another one.

This problem is caused by the way IRC server-client communication works. When you open Pesterchum and get on your chumhandle, you tell the server you are now using that name. After that, you and the server are constantly sending small messages back and forth every 30seconds or so (pings and pongs). By constantly receiving message from you, the server knows that you are still online and using your name. When your computer crashes or you suddenly lose connection to the server (your internet dies?), your client never gets a chance to inform the server that you are logging off. This causes the server to think you are still online, using your chumhandle. After a short time the server notices that you have stopped sending it messages and aren't responding to any messages it sends to you. It's at this point the server realizes that you're no longer online and frees up your name for use.

In short: Wait 5-10 minutes, the server will free up your chumhandle, and you'll be able to use it again.
